sos/plugins/kubernetes.py

```diff
@@ -22,33 +23,101 @@ class kubernetes(Plugin, RedHatPlugin):
     """Kubernetes plugin
     """
 
-    option_list = [("podslog", "capture logs for pods", 'slow', False)]
+    # Red Hat Atomic Platform and OpenShift Enterprise use the
+    # atomic-openshift-master package to provide kubernetes
+    packages = ('kubernetes', 'atomic-openshift-master')
```

Comment:
> In a containerized install there are no packages visible from the host,

That certainly isn't the case for Red Hat Atomic (or it wasn't when it was launched).

I understood that you were trying to extend the existing 'superspection' case (use of a super-privileged container to inspect the host), but based on:

> The atomic-openshift-master package was installed inside the image where this container is running from
> (openshift3/ose). You are running sosreport from another container.

It seems like you are actually trying to implement the 3rd use case that we'd discussed with the container folk, but in a single plugin. Iirc we termed this one 'extrospection' - using one container to inspect another. We discussed this in brief a couple of years ago but set it to one side as nobody at Red Hat had a use-case for it at the time.

There is currently no support for this in sos and it will almost certainly need core changes for it to work properly - it is roughly speaking a more complex version of what we now do for sysroot operation.

It would really help to have a clear plan of what you're trying to achieve, in terms of what contexts you're working with, what commands need to run in those contexts and where the file system data should be collected from.

I know the original patch from Jake was for e.g. trying to run commands that would only exist in the host in the chroot or vice-versa.